sdk/test-utils/recorder-new/src/core-v2-recorder.ts

```diff
@@ -224,6 +225,15 @@ export class TestProxyHttpClient {
     }
   }
 
+  /**
+   * Sets the matcher for the current recording to the matcher specified.
+   */
+  async setMatcher(matcher: Matcher): Promise<void> {
+    if (this.httpClient && this.mode === "playback") {
```

Comment:
I looked at the constructor and looks like if this.mode === "playback" either httpClient is set or the constructor throws. Should we throw here `if (this.mode === "playback") && !this.httpClient)` with a good message?

If not, what happens if we don't set the matcher? Does it just fail downstream somewhere due to some expectation? Not familiar with the code at all so just curious

Reply:
It's probably safe to assume that the `httpClient` is set if we're in playback mode and so we can simplify the logic here. Think I was just being a bit defensive in the check here.

The matcher doesn't actually need to be set at all, it's totally optional. When it is set, it changes the default behavior for how recordings are matched to a request during playback: using `HeaderlessMatcher` makes the test proxy ignore any differences between the recording and playback in the request headers and `BodilessMatcher` makes the proxy ignore any differences in the request body itself.